packages/core/test/state.test.ts

```diff
@@ -905,6 +905,27 @@ describe('State', () => {
 
       expect(nextState.can({ type: 'NEXT' })).toBeTruthy();
     });
+
+    // https://github.com/statelyai/xstate/issues/3096
+    it('should work with restored state', () => {
+      const machine = createMachine({
+        on: {
+          TOGGLE: {
+            actions: 'something'
+          }
+        }
+      });
+
+      const state = State.create(
+        JSON.parse(JSON.stringify(machine.initialState))
+      );
```

Comment:
this feels rather irrelevant for this test case - if we want to test that "machineless" state returns `false` then that should be moved to a dedicated test